core/controllers/question_editor.py

```diff
@@ -115,24 +115,28 @@ class QuestionSkillLinkHandler(base.BaseHandler):
     """A handler for linking and unlinking questions to or from a skill."""
 
     @acl_decorators.can_manage_question_skill_status
-    def put(self, question_id, skill_id):
-        """Updates the difficulty of the question with respect to the given
-        skill.
+    def put(self, question_id):
+        """Updates the QuestionSkillLink models with respect to the given
+        question.
         """
-        new_difficulty = float(self.payload.get('new_difficulty'))
-
-        question_services.update_question_skill_link_difficulty(
-            question_id, skill_id, new_difficulty)
+        if self.payload.get('action') == 'update_difficulty':
+            new_difficulty = float(self.payload.get('new_difficulty'))
+            skill_id = self.payload.get('skill_id')
```

Comment:
Make sure to have a test that confirms a 400 error is raised if skill_id is not passed in.

Reply:
Done